### PR TITLE
Fix model.evaluate don't have output

### LINKED ISSUE #986: Keras model.evaluate does not have output.
The `model.evaluate` in keras should output the metric of the model to terminal but not.

### PR DESCRIPTION
**Pull request by Wanglongzhi2001**
Fixed #986

### CODE REVIEW

File: src/TensorFlowNET.Keras/Engine/Model.Evaluate.cs [hunk 1]
```diff
@@ -46,18 +55,31 @@ public partial class Model
...
+                //callbacks.on_epoch_begin(epoch);
                 // data_handler.catch_stop_iteration();
-                IEnumerable<(string, Tensor)> results = null;
+                IEnumerable<(string, Tensor)> logs = null;
+
                 foreach (var step in data_handler.steps())
                 {
-                    // callbacks.on_train_batch_begin(step)
-                    results = test_function(data_handler, iterator);
+                    callbacks.on_train_batch_begin(step);
```
Comment (AsakusaRinne) [inline src/TensorFlowNET.Keras/Engine/Model.Evaluate.cs:75]: Should this be `on_test_batch_begin`?

Comment (Wanglongzhi2001) [reply src/TensorFlowNET.Keras/Engine/Model.Evaluate.cs:75]: > Should this be `on_test_batch_begin`?

sry, there are some mistakes happened, I will correct it.

**Comment by Wanglongzhi2001** [comment]
I'm sorry, but I would like to ask about the reason.

File: src/TensorFlowNET.Keras/Engine/Model.Evaluate.cs [hunk 2]
```diff
@@ -46,18 +55,32 @@ public partial class Model
...
+                    var end_step = step + data_handler.StepIncrement;
+                    callbacks.on_test_batch_end(end_step, logs);
                 }
             }
+            Console.WriteLine();
+            GC.Collect();
+            GC.WaitForPendingFinalizers();
         }
 
         public KeyValuePair<string, float>[] evaluate(IDatasetV2 x)
```
Comment (AsakusaRinne) [inline src/TensorFlowNET.Keras/Engine/Model.Evaluate.cs:86]: This method is an overload of the method above, which should have a similar implementation. I think maybe adding an `EvaluateInternal` is better, just like `FitInternal`.

**Review by AsakusaRinne — APPROVED** [review]

**Review by Oceania2018 — APPROVED** [review]
